edb/server/compiler/compiler.py

```diff
@@ -415,17 +421,17 @@ def _compile_ql_query(
                     named=named)
                 if array_params:
                     in_array_backend_tids = {p[0]: p[1] for p in array_params}
+                in_type_args = {name: name in optional_params
```

Comment:
Could we also use a dataclass here so that we don't have to change it again once we implement the `SET OF` modifier?

Reply:
Fixed

Reply:
Also moved array tids to the structure